subsys/bluetooth/audio/csip_set_member.c

```diff
@@ -202,29 +213,32 @@ int bt_csip_set_member_generate_rsi(const struct bt_csip_set_member_svc_inst *sv
 				    uint8_t rsi[BT_CSIP_RSI_SIZE])
 {
 	int res = 0;
-	uint32_t prand;
-	uint32_t hash;
+	uint8_t prand[BT_CSIP_SIH_PRAND_SIZE];
+	uint8_t hash[BT_CSIP_SIH_HASH_SIZE];
 
 	if (IS_ENABLED(CONFIG_BT_CSIP_SET_MEMBER_TEST_SAMPLE_DATA)) {
 		/* prand is from the sample data from A.2 in the CSIS spec */
-		prand = 0x69f563;
+		prand[0] = 0x63;
+		prand[1] = 0xf5;
+		prand[2] = 0x69;
```

Comment:
What you are doing here is essentially storing the prand as little-endian, and you can just use sys_put_le24. 

Since you are storing the value as little-endian, you can also just use sys_get_le24 in `generate_prand` when you do the checks for `has_ones` and `has_zeroes` and keep the old `if (*dest != 0 && *dest != 0x3FFFFF) {` check.

Reply:
I was specifically trying to avoid using uint32_t, that's why its stored in an array now :)
Sorry I cant remember right now, but one of the macros will cause the value to be "right-aligned" in uint32, i.e. prepending the value with an octet of zeroes. The idea with using arrays is that the solution is uniform for both system, and does not require a ton of special handling.

I do agree that the validation check in this case does not look as elegant, but I would argue that the overall solution has benefits.

Reply:
Sorry, it wasn't one of the macros, it was the bt_rand that was the issue:
`int bt_rand(void *buf, size_t len)`

Here an uint32t was used, which on LE endian systems would put all values aligned to the the lower indices- but on BE systems they would be aligned in the top indices, prepended with a 0 at index 0.

Reply:
Ah, yes, that makes sense. `res = bt_rand(dest, BT_CSIP_SIH_PRAND_SIZE);` should have been `res = bt_rand(dest, sizeof(*dest))` so that it used the entire 32-bits. 

The `*dest &= 0x3FFFFF;` further down would then have ensure that only the expected bits were set.

Reply:
I still think this is easier to read if we just do
```
sys_put_le24(0x69f563, prand);
```